crates/client/rpc/src/versions/v0_8_0/methods/ws/lib.rs

```
@@ -0,0 +1,370 @@
+use crate::{
+    errors::{ErrorExtWs, OptionExtWs, StarknetWsApiError},
+    versions::v0_8_0::StarknetWsRpcApiV0_8_0Server,
+};
+
+use super::BLOCK_PAST_LIMIT;
+
+#[jsonrpsee::core::async_trait]
+impl StarknetWsRpcApiV0_8_0Server for crate::Starknet {
+    async fn subscribe_new_heads(
+        &self,
+        pending: jsonrpsee::PendingSubscriptionSink,
+        block_id: starknet_core::types::BlockId,
+    ) -> jsonrpsee::core::SubscriptionResult {
+        let sink = pending.accept().await.or_internal_server_error("Failed to establish websocket connection")?;
+
+        let mut block_n = match block_id {
+            starknet_core::types::BlockId::Number(block_n) => {
+                let err = || format!("Failed to retrieve block info for block {block_n}");
+                let block_latest = self
+                    .backend
+                    .get_block_n(&mp_block::BlockId::Tag(mp_block::BlockTag::Latest))
+                    .or_else_internal_server_error(err)?
+                    .ok_or_else_internal_server_error(err)?;
```

None here happens when there are no blocks in DB, which should not internal server error, it should return a soft error kinda like StarknetApiError::NoBlocks in the standard rpc.

---

Fixed